README.md

```diff
@@ -3,8 +3,132 @@ A language server for Hoon.
 Install with:
 
 ```
-npm install -g hoon-language-server
+npm install -g @urbit/hoon-language-server
 ```
 
-Make sure you have a fake ~zod running on localhost:8080 and you've installed a
-Hoon plugin for your editor, like hoon.vim.
+This installs `hoon-language-server` which allows your code editor to communicate with an urbit ship running locally on your development machine.
+
+The default options are: `-p 80 -d 0 -u http://localhost -s zod -c lidlut-tabwed-pillex-ridrup`
+
+- `-p` is the port your ship is accessible on over HTTP
+- `-d` is the delay for running didSave events
+- `-u` is the URL of your ship
+- `-s` is the `@p` of your ship (without a sig)
+- `-c` is the `+code` of the running urbit (also without a sig)
+
+## Urbit Setup
+
+You must have a fake ship running and it must be running the hoon language server.  To create and start a fake `~zod`:
+
+```
+urbit -F zod -c zod
+```
+
+In the urbit dojo, start the language server:
+
+```
+|start %language-server
+```
+
+If you need the passcode for your ship, enter this in dojo:
+
+```
++code
+```
+
+To start the same ship again in the future just run:
+
+```
+urbit zod
+```
+
+in the same directory you created it in.
+
+## Editor Setup
+
+Your code editor now needs to use `hoon-language-server` as an LSP provider.  There are plugins for common editors that add syntax highlighting, etc. and help connect to the language server.
+
+### VSCode
+
+ * [hoon-vscode](https://github.com/famousj/hoon-vscode)
+ * [hoon-assist-vscode](https://github.com/urbit/hoon-assist-vscode)
+
+### Emacs
+
+ * [hoon-mode.el](https://github.com/urbit/hoon-mode.el)
+
+### Vim
+
+ * [hoon.vim](https://github.com/urbit/hoon.vim)
+
+`hoon.vim` does not use the language server itself, but the github page describes a setup using `vim-lsp`.
+
+### Neovim
+
+Neovim users should use [hoon.vim](https://github.com/urbit/hoon.vim) with one of the following LSP setups:
+
+#### Native LSP
+
+Install [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) to simplify your local configuration.  To use the [default configuration](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.txt#hoon_ls), add the following to `init.lua`:
+
+```
+require'lspconfig'.hoon_ls.setup{}
+```
+
+To modify the default options use:
+
+```
+require'lspconfig'.hoon_ls.setup{
+  cmd = {"hoon-language-server" "-p", "80", "-d", "500", -s "zod", "-u", "http://localhost", "-c" "lidlut-tabwed-pillex-ridrup"},
+  filetypes = {"hoon"},
+  single_file_support = true
```

Comment:
That is part of the default configuration. 
I suggest to only use the "cmd" command as example here. All other options are more for advanced use cases.

```suggestion
  cmd = {"hoon-language-server" "-p", "80", "-d", "500", -s "zod", "-u", "http://localhost", "-c" "lidlut-tabwed-pillex-ridrup"}
```

Reply:
not really useful unless we want to add support for udon files or something.